### ISSUE TITLE
JDK 21.0.2 + VT + System.out.println + Gradle (org.gradle.internal.io.LinePerThreadBufferingOutputStream) = chance of hang-up

### ISSUE THREAD

**Issue by magicprinc** [gradle/gradle#27779]
### Current Behavior

I can't reproduce it in a small demo, but it is real.

JDK 21.0.2 + Virtual Threads (more than 100) + System.out.println(inside VT) + Gradle test runner (org.gradle.internal.io.LinePerThreadBufferingOutputStream)
and … from time to time it hangs up.

If you change any part: IDEA runner instead of Gradle or System.out.write instead of println or platform threads instead of virtual threads - everything works without problems.

It is blocked so hard: I can't make dump (Visual VM and command line tools hangs up in process).
So does Thread.getStackTrace()

The only thing I could capture is: Thread's toString

They all are in this state:
```
VirtualThread[#64,VT-3]/blocked@ForkJoinPool-1-worker-21
VirtualThread[#65,VT-4]/blocked@ForkJoinPool-1-worker-31
VirtualThread[#66,VT-5]/blocked@ForkJoinPool-1-worker-41
```

`-Djdk.tracePinnedThreads=full` doesn't show anything

### Expected Behavior

It must not hang up

### Steps to Reproduce

JDK 21.0.2 + Virtual Threads (more than 100) + System.out.println(inside VT) + Gradle test runner 
+ 
`-XX:ActiveProcessorCount=64` or more

### Gradle version

8.5

### Your Environment (optional)

JDK 21.0.2, Windows 10

**Comment by magicprinc** [comment]
`Logback ConsoleAppender` works: it uses `System.out.write():
I have changed `println` to `write` and: no more deadlocking!
So the problem is somewhere in the depths of your `LinePerThreadBufferingOutputStream.println()` implementation.

**Comment by cobexer** [comment]
Thank you for your interest in Gradle!

This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.
***
@gradle/bt-core-runtime-maintainers would we do something about this without a reproducer?
***
If you find a way to provide a reproducer it would help a lot!

The ideal way to provide a reproducer is to leverage our [reproducer template](https://github.com/gradle/gradle-issue-reproducer). You can also use [Gradle Project Replicator](https://github.com/android/project-replicator) to reproduce the structure of your project.

Also adding more details on what the build is supposed to do and how may also help us.
How do VirtualThreads come into play here?

**Comment by magicprinc** [comment]
> How do VirtualThreads come into play here?
No idea :-)
If I use VT thread poll then it blocks
If I use normal (platform) thread pool then it works

**Comment by mlopatkin** [comment]
>@gradle/bt-core-runtime-maintainers would we do something about this without a reproducer?

It is a bit tricky to proceed without a thread dump, but probably possible. Most likely, the situation is similar to what is described in [this article](https://blog.ydb.tech/how-we-switched-to-java-21-virtual-threads-and-got-deadlock-in-tpc-c-for-postgresql-cca2fe08d70b) - all carrier threads wait for synchronized and output connection, while a virtual thread holding the connection lock is unmounted and has no free carrier thread to mount on.

It is probably solvable (e.g. we can start with replacing `synchronized` blocks with `ReentrantLock` in the `LinePerThreadBufferingOutputStream`), but I cannot estimate how deep this rabbit hole is. Gradle doesn't use virtual threads and most plugins likely don't either, so I don't see this issue as a high priority.

**Comment by henryju** [comment]
Hi, I have the impression of being affected by a similar issue (Gradle stuck on LinePerThreadBufferingOutputStream
 println), but this is with Java 17. Let me know if I should open a separate issue.

JVM: OpenJDK 64-Bit Server VM (17.0.9+9-Ubuntu-122.04)
Gradle: 8.3

I cannot reproduce with Gradle 8.6, but maybe I am just lucky.

I managed to collect a thread dump:

```
"Test worker" #1 prio=5 os_prio=0 cpu=10497,63ms elapsed=396,93s tid=0x00007f21b401c090 nid=0x5a260 waiting on condition  [0x00007f21bbdfe000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.9/Native Method)
        - parking to wait for  <0x00000000e0b6ab40> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.9/LockSupport.java:252)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@17.0.9/AbstractQueuedSynchronizer.java:1672)
        at java.util.concurrent.ThreadPoolExecutor.awaitTermination(java.base@17.0.9/ThreadPoolExecutor.java:1464)
        at org.gradle.internal.concurrent.AbstractManagedExecutor.stop(AbstractManagedExecutor.java:86)
        at org.gradle.internal.concurrent.DefaultExecutorFactory$TrackedManagedExecutor.stop(DefaultExecutorFactory.java:116)
        at org.gradle.internal.concurrent.AbstractManagedExecutor.stop(AbstractManagedExecutor.java:76)
        at org.gradle.internal.remote.internal.hub.MessageHub.stop(MessageHub.java:225)
        at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:106)
        - locked <0x00000000ff751208> (a org.gradle.internal.concurrent.CompositeStoppable)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection.stop(MessageHubBackedObjectConnection.java:150)
        at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:106)
        - locked <0x00000000ff750c60> (a org.gradle.internal.concurrent.CompositeStoppable)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:117)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

   Locked ownable synchronizers:
        - None

"/127.0.0.1:59742 to /127.0.0.1:35013 workers" #24 prio=5 os_prio=0 cpu=8,52ms elapsed=393,66s tid=0x00007f21b4b0d670 nid=0x5a36f waiting on condition  [0x00007f218d4f5000]
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.9/Native Method)
        - parking to wait for  <0x00000000e03fd460> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(java.base@17.0.9/LockSupport.java:341)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(java.base@17.0.9/AbstractQueuedSynchronizer.java:506)
        at java.util.concurrent.ForkJoinPool.unmanagedBlock(java.base@17.0.9/ForkJoinPool.java:3465)
        at java.util.concurrent.ForkJoinPool.managedBlock(java.base@17.0.9/ForkJoinPool.java:3436)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@17.0.9/AbstractQueuedSynchronizer.java:1623)
        at org.gradle.internal.remote.internal.hub.queue.EndPointQueue.take(EndPointQueue.java:49)
        at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:403)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.9/ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.9/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.9/Thread.java:840)

   Locked ownable synchronizers:
        - <0x00000000e03fd848> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"/127.0.0.1:59742 to /127.0.0.1:35013 workers Thread 2" #25 prio=5 os_prio=0 cpu=1787,88ms elapsed=393,51s tid=0x00007f21b4b3a6f0 nid=0x5a372 runnable  [0x00007f218d3f5000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.EPoll.wait(java.base@17.0.9/Native Method)
        at sun.nio.ch.EPollSelectorImpl.doSelect(java.base@17.0.9/EPollSelectorImpl.java:118)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(java.base@17.0.9/SelectorImpl.java:129)
        - locked <0x00000000e0faec78> (a sun.nio.ch.Util$2)
        - locked <0x00000000e0faec28> (a sun.nio.ch.EPollSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(java.base@17.0.9/SelectorImpl.java:146)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.waitForWriteBufferToDrain(SocketConnection.java:296)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.writeBufferToChannel(SocketConnection.java:270)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.write(SocketConnection.java:253)
        at com.esotericsoftware.kryo.io.Output.flush(Output.java:163)
        at com.esotericsoftware.kryo.io.Output.require(Output.java:142)
        at com.esotericsoftware.kryo.io.Output.writeLong(Output.java:501)
        at org.gradle.internal.serialize.kryo.KryoBackedEncoder.writeLong(KryoBackedEncoder.java:53)
        at org.gradle.api.internal.tasks.testing.worker.TestEventSerializer$IdSerializer.write(TestEventSerializer.java:100)
        at org.gradle.api.internal.tasks.testing.worker.TestEventSerializer$IdSerializer.write(TestEventSerializer.java:91)
        at org.gradle.internal.serialize.DefaultSerializerRegistry$TaggedTypeSerializer.write(DefaultSerializerRegistry.java:171)
        at org.gradle.internal.remote.internal.hub.DefaultMethodArgsSerializer$ArraySerializer.write(DefaultMethodArgsSerializer.java:80)
        at org.gradle.internal.remote.internal.hub.DefaultMethodArgsSerializer$ArraySerializer.write(DefaultMethodArgsSerializer.java:61)
        at org.gradle.internal.remote.internal.hub.MethodInvocationSerializer$MethodInvocationWriter.writeArgs(MethodInvocationSerializer.java:78)
        at org.gradle.internal.remote.internal.hub.MethodInvocationSerializer$MethodInvocationWriter.write(MethodInvocationSerializer.java:74)
        at org.gradle.internal.remote.internal.hub.MethodInvocationSerializer$MethodInvocationWriter.write(MethodInvocationSerializer.java:58)
        at org.gradle.internal.serialize.kryo.TypeSafeSerializer$2.write(TypeSafeSerializer.java:47)
        at org.gradle.internal.remote.internal.hub.InterHubMessageSerializer$MessageWriter.write(InterHubMessageSerializer.java:104)
        at org.gradle.internal.remote.internal.hub.InterHubMessageSerializer$MessageWriter.write(InterHubMessageSerializer.java:88)
        at org.gradle.internal.remote.internal.inet.SocketConnection.dispatch(SocketConnection.java:122)
        at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionDispatch.run(MessageHub.java:328)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.9/ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.9/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.9/Thread.java:840)

   Locked ownable synchronizers:
        - <0x00000000e03fdb88> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"/127.0.0.1:59742 to /127.0.0.1:35013 workers Thread 3" #26 prio=5 os_prio=0 cpu=12,24ms elapsed=393,50s tid=0x00007f21b4b3b4e0 nid=0x5a373 runnable  [0x00007f218d2f5000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.EPoll.wait(java.base@17.0.9/Native Method)
        at sun.nio.ch.EPollSelectorImpl.doSelect(java.base@17.0.9/EPollSelectorImpl.java:118)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(java.base@17.0.9/SelectorImpl.java:129)
        - locked <0x00000000e03fe8b0> (a sun.nio.ch.Util$2)
        - locked <0x00000000e03fe860> (a sun.nio.ch.EPollSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(java.base@17.0.9/SelectorImpl.java:146)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketInputStream.read(SocketConnection.java:185)
        at com.esotericsoftware.kryo.io.Input.fill(Input.java:146)
        at com.esotericsoftware.kryo.io.Input.require(Input.java:178)
        at com.esotericsoftware.kryo.io.Input.readByte(Input.java:295)
        at org.gradle.internal.serialize.kryo.KryoBackedDecoder.readByte(KryoBackedDecoder.java:88)
        at org.gradle.internal.remote.internal.hub.InterHubMessageSerializer$MessageReader.read(InterHubMessageSerializer.java:64)
        at org.gradle.internal.remote.internal.hub.InterHubMessageSerializer$MessageReader.read(InterHubMessageSerializer.java:52)
        at org.gradle.internal.remote.internal.inet.SocketConnection.receive(SocketConnection.java:81)
        at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionReceive.run(MessageHub.java:270)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.9/ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.9/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.9/Thread.java:840)

   Locked ownable synchronizers:
        - <0x00000000e0906e08> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"pool-12-thread-1" #108 prio=5 os_prio=0 cpu=15915,67ms elapsed=367,58s tid=0x00007f2078006510 nid=0x5abef runnable  [0x00007f218c1f1000]
   java.lang.Thread.State: RUNNABLE
        at java.nio.charset.CharsetEncoder.encode(java.base@17.0.9/CharsetEncoder.java:585)
        at sun.nio.cs.StreamEncoder.implWrite(java.base@17.0.9/StreamEncoder.java:293)
        at sun.nio.cs.StreamEncoder.implWrite(java.base@17.0.9/StreamEncoder.java:282)
        at sun.nio.cs.StreamEncoder.write(java.base@17.0.9/StreamEncoder.java:132)
        - locked <0x00000000ff748290> (a java.io.OutputStreamWriter)
        at java.io.OutputStreamWriter.write(java.base@17.0.9/OutputStreamWriter.java:205)
        at java.io.BufferedWriter.flushBuffer(java.base@17.0.9/BufferedWriter.java:120)
        - locked <0x00000000ff748290> (a java.io.OutputStreamWriter)
        at java.io.PrintStream.write(java.base@17.0.9/PrintStream.java:698)
        - locked <0x00000000ff746188> (a java.io.PrintStream)
        at java.io.PrintStream.print(java.base@17.0.9/PrintStream.java:863)
        at org.gradle.internal.io.LinePerThreadBufferingOutputStream.println(LinePerThreadBufferingOutputStream.java:239)
        - locked <0x00000000e0b53340> (a org.gradle.internal.io.LinePerThreadBufferingOutputStream)
        at io.mycompny.myproject.worker.shared.QueueSubscriberTest$LongWorker.execute(QueueSubscriberTest.java:664)
        at io.mycompny.myproject.worker.shared.QueueSubscriber.lambda$processMessage$0(QueueSubscriber.java:158)
        at io.mycompny.myproject.worker.shared.QueueSubscriber$$Lambda$662/0x00007f214040b4a0.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(java.base@17.0.9/FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.9/ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.9/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.9/Thread.java:840)

   Locked ownable synchronizers:
        - <0x00000000e0d74610> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"pool-45-thread-1" #269 prio=5 os_prio=0 cpu=14834,31ms elapsed=364,20s tid=0x00007f206c001f30 nid=0x5acf4 waiting for monitor entry  [0x00007f213edfd000]
   java.lang.Thread.State: BLOCKED (on object monitor)
        at org.gradle.internal.io.LinePerThreadBufferingOutputStream.println(LinePerThreadBufferingOutputStream.java:238)
        - waiting to lock <0x00000000e0b53340> (a org.gradle.internal.io.LinePerThreadBufferingOutputStream)
        at io.mycompny.myproject.worker.shared.QueueSubscriberTest$LongWorker.execute(QueueSubscriberTest.java:664)
        at io.mycompny.myproject.worker.shared.QueueSubscriber.lambda$processMessage$0(QueueSubscriber.java:158)
        at io.mycompny.myproject.worker.shared.QueueSubscriber$$Lambda$662/0x00007f214040b4a0.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(java.base@17.0.9/FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@17.0.9/ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@17.0.9/ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(java.base@17.0.9/Thread.java:840)

   Locked ownable synchronizers:
        - <0x00000000f1f0fdc8> (a java.util.concurrent.ThreadPoolExecutor$Worker)

```

**Comment by henryju** [comment]
I was affected by the issue multiple times in a row, yes. I forgot to mention we are using `org.gradle.parallel=true`

**Comment by AlexeyKuznetsov-DD** [comment]
Hi!
Recently I got very similar issue on GitLab CI, it is not 100% reproducible, maybe 5-7% chance only.
I got the following stack trace with Gradle 8.14.4 and demon JVM 21, 
```
------------------------------------------------------------
Gradle 8.14.4
------------------------------------------------------------
Build time:    2026-01-23 16:30:23 UTC
Revision:      ad5ff774b4b0e9a8a0cf1a14ca70d7230003c3ad
Kotlin:        2.0.21
Groovy:        3.0.25
Ant:           Apache Ant(TM) version 1.10.15 compiled on August 25 2024
Launcher JVM:  1.8.0_472 (Temurin 25.472-b08)
Daemon JVM:    Compatible with Java 21, any vendor, nativeImageCapable=false (from gradle/gradle-daemon-jvm.properties)
OS:            Linux 6.8.0 amd64
```
Stacktrace:
```
"Test worker" #1 prio=5 os_prio=0 cpu=8677.27ms elapsed=1142.29s tid=0x00007f353c02d000 nid=0x9705 waiting for monitor entry  [0x00007f354233e000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at org.gradle.internal.io.LinePerThreadBufferingOutputStream.println(LinePerThreadBufferingOutputStream.java:237)
	- waiting to lock <0x00000000c062fa70> (a org.gradle.internal.io.LinePerThreadBufferingOutputStream)
	at org.slf4j.impl.SimpleLogger.write(SimpleLogger.java:318)
	at org.slf4j.impl.SimpleLogger.log(SimpleLogger.java:295)
	at org.slf4j.impl.SimpleLogger.formatAndLog(SimpleLogger.java:355)
	at org.slf4j.impl.SimpleLogger.debug(SimpleLogger.java:446)
	at datadog.trace.api.Config.<init>(Config.java:2982)
	at datadog.trace.api.Config.<init>(Config.java:1334)
	at datadog.trace.api.Config.<init>(Config.java:1330)
	at jdk.internal.reflect.GeneratedConstructorAccessor18.newInstance(java.base@11.0.30/Unknown Source)
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(java.base@11.0.30/DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(java.base@11.0.30/Constructor.java:490)
	at jdk.internal.reflect.GeneratedMethodAccessor45.invoke(java.base@11.0.30/Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.30/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.30/Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSite.invoke(PojoMetaMethodSite.java:186)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:130)
	at datadog.trace.test.util.DDSpecification.rebuildConfig(DDSpecification.groovy:245)
	- locked <0x00000000c0fc4470> (a java.lang.Class for datadog.trace.test.util.DDSpecification)
	at datadog.trace.agent.test.InstrumentationSpecification.super$4$rebuildConfig(InstrumentationSpecification.groovy)
	at jdk.internal.reflect.GeneratedMethodAccessor51.invoke(Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.30/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.30/Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1254)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuperN(ScriptBytecodeAdapter.java:145)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnSuper0(ScriptBytecodeAdapter.java:165)
	at datadog.trace.agent.test.InstrumentationSpecification.rebuildConfig(InstrumentationSpecification.groovy:483)
	at datadog.trace.test.util.DDSpecification.setup(DDSpecification.groovy:152)
	at jdk.internal.reflect.GeneratedMethodAccessor67.invoke(Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.30/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.30/Method.java:566)
	at org.spockframework.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:277)
	at org.spockframework.runtime.model.MethodInfo.lambda$new$0(MethodInfo.java:49)
	at org.spockframework.runtime.model.MethodInfo$$Lambda$270/0x0000000100395040.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:325)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:321)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:321)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:321)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:321)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:321)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.doRunSetup(PlatformSpecRunner.java:321)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSetup$7(PlatformSpecRunner.java:305)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$668/0x0000000100611c40.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:299)
	at org.spockframework.runtime.PlatformSpecRunner.runSetup(PlatformSpecRunner.java:294)
	at org.spockframework.runtime.IterationNode.before(IterationNode.java:41)
	at org.spockframework.runtime.IterationNode.before(IterationNode.java:13)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:154)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$451/0x00000001003f9c40.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$450/0x00000001003f9840.invoke(Unknown Source)
	at org.spockframework.runtime.SpockNode.sneakyInvoke(SpockNode.java:40)
	at org.spockframework.runtime.IterationNode.lambda$around$0(IterationNode.java:68)
	at org.spockframework.runtime.IterationNode$$Lambda$665/0x0000000100612040.run(Unknown Source)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunIteration$5(PlatformSpecRunner.java:244)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$666/0x0000000100612440.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runIteration(PlatformSpecRunner.java:226)
	at org.spockframework.runtime.IterationNode.around(IterationNode.java:68)
	at org.spockframework.runtime.IterationNode.around(IterationNode.java:13)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$449/0x00000001003f9440.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$DefaultDynamicTestExecutor.execute(NodeTestTask.java:232)
	at org.spockframework.runtime.SimpleFeatureNode.execute(SimpleFeatureNode.java:58)
	at org.spockframework.runtime.SimpleFeatureNode.execute(SimpleFeatureNode.java:19)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:157)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$451/0x00000001003f9c40.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$450/0x00000001003f9840.invoke(Unknown Source)
	at org.spockframework.runtime.SpockNode.sneakyInvoke(SpockNode.java:40)
	at org.spockframework.runtime.FeatureNode.lambda$around$0(FeatureNode.java:73)
	at org.spockframework.runtime.FeatureNode$$Lambda$650/0x0000000100616440.run(Unknown Source)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunFeature$4(PlatformSpecRunner.java:208)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$651/0x0000000100615840.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runFeature(PlatformSpecRunner.java:199)
	at org.spockframework.runtime.FeatureNode.around(FeatureNode.java:73)
	at org.spockframework.runtime.FeatureNode.around(FeatureNode.java:30)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$449/0x00000001003f9440.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$455/0x00000001003fac40.accept(Unknown Source)
	at java.util.ArrayList.forEach(java.base@11.0.30/ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$451/0x00000001003f9c40.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$450/0x00000001003f9840.invoke(Unknown Source)
	at org.spockframework.runtime.SpockNode.sneakyInvoke(SpockNode.java:40)
	at org.spockframework.runtime.SpecNode.lambda$around$0(SpecNode.java:72)
	at org.spockframework.runtime.SpecNode$$Lambda$568/0x00000001004e0c40.run(Unknown Source)
	at org.spockframework.runtime.PlatformSpecRunner.lambda$createMethodInfoForDoRunSpec$0(PlatformSpecRunner.java:66)
	at org.spockframework.runtime.PlatformSpecRunner$$Lambda$569/0x00000001004e1040.invoke(Unknown Source)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:157)
	at org.spockframework.runtime.PlatformSpecRunner.invokeRaw(PlatformSpecRunner.java:442)
	at org.spockframework.runtime.PlatformSpecRunner.invoke(PlatformSpecRunner.java:425)
	at org.spockframework.runtime.PlatformSpecRunner.runSpec(PlatformSpecRunner.java:59)
	at org.spockframework.runtime.SpecNode.around(SpecNode.java:72)
	at org.spockframework.runtime.SpecNode.around(SpecNode.java:12)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$449/0x00000001003f9440.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$455/0x00000001003fac40.accept(Unknown Source)
	at java.util.ArrayList.forEach(java.base@11.0.30/ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$451/0x00000001003f9c40.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$450/0x00000001003f9840.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$449/0x00000001003f9440.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.executeEngine(EngineExecutionOrchestrator.java:230)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.failOrExecuteEngine(EngineExecutionOrchestrator.java:204)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:172)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:101)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:64)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator$$Lambda$414/0x00000001003e5040.accept(Unknown Source)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:150)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:63)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:109)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:91)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$execute$1(InterceptingLauncher.java:39)
	at org.junit.platform.launcher.core.InterceptingLauncher$$Lambda$135/0x00000001001e6440.proceed(Unknown Source)
	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
	at org.junit.platform.launcher.core.InterceptingLauncher.execute(InterceptingLauncher.java:38)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:124)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:94)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:63)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@11.0.30/Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@11.0.30/NativeMethodAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.30/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.30/Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
	at com.sun.proxy.$Proxy4.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:132)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoader
```

**Comment by AlexeyKuznetsov-DD** [comment]
Gradle 8.14.4 is running with parameters: `--build-cache --stacktrace --no-daemon --parallel --max-workers=4`

Interesting, today I got another hanged test, but with a bit different thread dump:
```
"Test worker" #1 prio=5 os_prio=0 cpu=9462.26ms elapsed=1143.01s tid=0x00007f4c7402d000 nid=0x95e3 waiting on condition  [0x00007f4c79f6a000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.30/Native Method)
	- parking to wait for  <0x00000000c0643a90> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.30/LockSupport.java:234)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.30/AbstractQueuedSynchronizer.java:2123)
	at java.util.concurrent.ThreadPoolExecutor.awaitTermination(java.base@11.0.30/ThreadPoolExecutor.java:1454)
	at org.gradle.internal.concurrent.AbstractManagedExecutor.stop(AbstractManagedExecutor.java:87)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$TrackedManagedExecutor.stop(DefaultExecutorFactory.java:116)
	at org.gradle.internal.concurrent.AbstractManagedExecutor.stop(AbstractManagedExecutor.java:77)
	at org.gradle.internal.remote.internal.hub.MessageHub.stop(MessageHub.java:225)
	at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:114)
	- locked <0x00000000d4dbe2b0> (a org.gradle.internal.concurrent.CompositeStoppable)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection.stop(MessageHubBackedObjectConnection.java:150)
	at org.gradle.internal.concurrent.CompositeStoppable.stop(CompositeStoppable.java:114)
	- locked <0x00000000d4dbdfd8> (a org.gradle.internal.concurrent.CompositeStoppable)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:126)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

**Comment by AlexeyKuznetsov-DD** [comment]
Update on issue reported by me:

I’ve identified the root cause, and it turned out not to be a Gradle bug.

We had a test that would hang on CI with ~10% probability. Our initial hypothesis was that the issue was caused by an outdated `Logback` version. However, after switching to `slf4j-simple`, the test still hung — this time inside Gradle code while attempting to write to the console.

Given that:

- CI runs Gradle builds in parallel
- The environment is virtualized
- There are likely limits around concurrent console output

This pointed to console I/O as a contributing factor rather than the logging implementation itself.

The actual root cause was an infinite recursion during error handling inside the test, which caused excessive log output. Under CI conditions, this log flooding led to the observed hanging behavior.

Previous attempts (e.g., switching logging implementations) effectively masked the symptom rather than addressing the underlying problem. This is also supported by the fact that we run ~200k tests with `Logback` without any similar hanging issues.

After fixing the infinite recursion and stopping the excessive logging, the issue was fully resolved.

P.S. [Link](https://github.com/DataDog/dd-trace-java/pull/10535) to fixed issue in our code.
